ERCS/erc-XXXX.md

```diff
@@ -13,60 +13,64 @@ requires: 712, 1193
-* dApp maintainer are at risk of losing their trusted reputation if an attacker finds a viable MITM attack
+- Users cannot simply trust otherwise honest dapps, and are at risk of losing funds
+- Dapp maintainers are at risk of hurting their reputations if an attacker finds a viable MITM attack
 
 ## Specification
 
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
 ### Overview
 
-We are proposing to use the domain certificate of a dApp as a root of trust to establish a trust chain as follow:
+We propose to use the dapp's domain certificate of a root of trust to establish a trust chain as follow:
 
-1. end user's browser verifies the domain certificate and display appropriate warning if overtaken
-2. the DNS record of the dApp hosts a TXT field pointing to a URL where a JSON manifest is hosted (this file SHOULD be at a well known address such as <https://domain.com/.well-known/twit.json>)
-3. the config file contains an array of tuple `[alg, public key]`
-4. for secure interaction, a dApp would first call their backend to securely sign the payload with a private key
-5. the encrypted payload would be sent to the wallet via the `secureRequest` JSON api call
-6. the wallet would verify the signature before sending the payload to the `request` JSON api call
+1. The user's browser verifies the domain certificate and displays appropriate warnings if overtaken
+2. The DNS record of the dapp hosts a TXT field pointing to a URL where a JSON manifest is hosted
+   - This file SHOULD be at a well known address such as <https://domain.com/.well-known/twit.json>
+3. The config file contains an array of objects of the form `{ id, alg, publicKey }`
+4. For signed requests, the dapp first securely signs the payload with a private key, for example by submitting a request to its backend
+5. The original payload, signature, and public key id are sent to the wallet via the `wallet_signedRequest` RPC method
+6. The wallet verifies the signature before processing the request normally
 
 ### Wallet integration
 
 #### Key discovery
 
-Attested public keys are necessary for the chain of trust to be established. Since this is traditionally done via DNS certificate, we are proposing to anchor directly on the DNS record to point to a
-manifest file containing the keys. This is similar to DKIM, but the use of the configuration file provides more flexibility for future improvements, and support for multiple algo/key pairs.
+Attested public keys are necessary for the chain of trust to be established.
+Since this is traditionally done via DNS certificates, we propose the addition of a DNS record containing the public keys.
+This is similar to DKIM, but the use of the configuration file provides more flexibility for future improvements, and support for multiple algorithm and key pairs.
```

Comment:
nit: manifest vs configuration? manifest is readonly and made to be published, whereas config file are  editable and not necesarily public

Reply:
Changed to manifest in [7e72a2e](https://github.com/uni-guillaume/ERCs/pull/2/commits/7e72a2eb1cac851a12f74e4d77f9eb90a598a2c3)